tests/integration/test_clip.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+from utils.generators import mk_seq_array
+
+import cunumeric as num
+
+# cunumeric.clip(array, a_min, a_max, out=None, **kwargs)
+
+
+@pytest.mark.xfail
+def test_none_array():
+    expected_exc = TypeError
+    with pytest.raises(expected_exc):
+        np.clip(None, a_min=0, a_max=0)
+    with pytest.raises(expected_exc):
+        # cunumeric raises
+        # AttributeError: 'NoneType' object has no attribute 'clip'
+        num.clip(None, a_min=0, a_max=0)
+
+
+def test_empty_array():
+    res_np = np.clip([0], a_min=0, a_max=0)
+    res_num = num.clip([0], a_min=0, a_max=0)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+def test_value_none():
+    array = np.arange(0, 10)
+    expected_exc = ValueError
+    with pytest.raises(expected_exc):
+        # Numpy raises:
+        # ValueError: One of max or min must be given
+        np.clip(array, a_min=None, a_max=None)
+    with pytest.raises(expected_exc):
+        # cunumeric raises:
+        # TypeError: int() argument must be a string, a bytes-like object
+        # or a real number, not 'NoneType'
+        num.clip(array, a_min=None, a_max=None)
+
+
+@pytest.mark.xfail
+def test_amin_amax():
+    array = np.arange(0, 10)
+    res_np = np.clip(array, a_min=9, a_max=5)
+    res_num = num.clip(array, a_min=9, a_max=5)
+    # the result is different
+    # array = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    # res_np = array([5, 5, 5, 5, 5, 5, 5, 5, 5, 5])
+    # res_num = array([9, 9, 9, 9, 9, 9, 9, 9, 9, 5])
```

Comment:
cc @manopapad seems like a genuine bug? (not just different error case handling, etc)

Reply:
sorry, it doesn't work. So I keep it.